libfwupdplugin/fu-plugin.c

```diff
@@ -162,6 +163,43 @@ fu_plugin_set_name (FuPlugin *self, const gchar *name)
 	priv->name = g_strdup (name);
 }
 
+/**
+ * fu_plugin_get_mode_text:
+ * @self: A #FuPlugin
+ *
+ * Gets the plugin text indicating why it is degraded.
+ *
+ * Returns: a plugin mode_text, or %NULL for unknown.
+ *
+ * Since: 1.5.0
+ **/
+const gchar *
```

Comment:
So I was actually thinking this might want to be a fixed set of reasons that are translatable by clients.  `fwup-enums.h` can have indices to match them.  When new reasons come up from new plugins we can add to the enums.

Reply:
> new reasons come up from new plugins we can add to the enums

Very good idea. Will respin with a bitfield instead.